LanguageFeatures/Patterns/pattern_assignment_A05_t06.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A pattern on the left side of an assignment expression is used to
+/// destructure the assigned value. We extend expression:
+///
+/// expression        ::= patternAssignment
+///                     | // Existing productions...
+///
+/// patternAssignment ::= outerPattern '=' expression
+/// ...
+/// It is a compile-time error if:
+///
+/// An identifier in a variable pattern does not resolve to an assignable local
+/// variable or formal parameter. A variable is assignable if it is any of:
+///
+/// - Non-final
+/// - Final and definitely unassigned
+/// - Late final and not definitely assigned
+///
+/// @description Check that it is no compile-time error if an identifier in a
+/// variable pattern is late final and not definitely assigned
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns,records
+
+import "../../Utils/expect.dart";
+import "patterns_lib.dart";
+
+main() {
+  var a = 42;
+  Record r = ();
+  final int af1, af2, af3, af4, af5, af6, af7, af8, areaAsInt;
+  final Record rf;
+  final List<int> lf;
+
+  if (2 > 1) {
+    af1 = 0;
+    af2 = 0;
+    af3 = 0;
+    af4 = 0;
+    af5 = 0;
+    af6 = 0;
+    af7 = 0;
+    af8 = 0;
+    areaAsInt = 0;
+    rf = ();
+    lf = [0];
+  }
+
+  Expect.throws(() {
+    ((af1, _) && r) = (1, 2);
```

Comment:
It seems that this line should trigger a compile-time error. I wrote the following program to see the behavior of both the CFE and the analyzer in analogous situation without patterns, and both tools report an error.

```dart
test(void Function() f) => f();

main() {
  final int a;
  if (2 > 1) {
    a = 0;
  }
  test(() {
      a = 1;
  });
}
```

The output from the CFE.
```
$ xcodebuild/ReleaseX64/dart /tmp/foo.dart
/tmp/foo.dart:9:7: Error: Final variable 'a' might already be assigned at this point.
      a = 1;
      ^
```

The output from the Analyzer.
```
$ xcodebuild/ReleaseX64/dart analyze /tmp/foo.dart
Analyzing foo.dart...

  error - foo.dart:9:7 - The final variable 'a' can only be set once. Try making 'a' non-final. - assignment_to_final_local
   info - foo.dart:4:13 - The value of the local variable 'a' isn't used. Try removing the variable or using it. - unused_local_variable

2 issues found.
```

Similarly below.

Reply:
Test description clearly says about `late final` but the test had only `final`. Added `late`. With the `late` an example above produces a runtime error, not a compile-time. Thank you for noticing